### PR TITLE
Remove some catch alls

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

- Allow Interrupted Exceptions to float out of the web requests.
- Use `Type_Error` rather than Any when catching auto scoping resolving.
- Rename `Java_Exception` to `JException`

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

**Review by AdRiley — APPROVED** [review]

File: distribution/lib/Standard/AWS/0.0.0-dev/src/Database/Redshift/Redshift_Details.enso [hunk 1]
```diff
@@ -36,7 +37,7 @@ type Redshift_Details
        Attempt to resolve the constructor.
     resolve : Function -> Redshift_Details | Nothing
     resolve constructor =
-        Panic.catch Any (constructor:Redshift_Details) _->Nothing
+        Panic.catch Type_Error (constructor:Redshift_Details) _->Nothing
```
Comment (radeusgd) [inline distribution/lib/Standard/AWS/0.0.0-dev/src/Database/Redshift/Redshift_Details.enso:40]: Btw im wondering if we could extrac this into a common pattern:

```
resolve_constructor ctor x = Panic.catch Type_Error (ctor.from X) _->Nothing 
```
?

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data_Formatter.enso [hunk 2]
```diff
@@ -13,7 +13,7 @@ import project.Value_Type.Auto
 import project.Value_Type.Bits
 import project.Value_Type.Value_Type
 
-polyglot java import java.lang.Exception as Java_Exception
+polyglot java import java.lang.Exception as JException
```
Comment (radeusgd) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Data_Formatter.enso:16]: Just trying to understand: what was wrong with `Java_Exception` name?

**Review by radeusgd — APPROVED** [review]
Looks good, just some notes

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data_Formatter.enso (hunk 2 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Data_Formatter.enso:16]: Makes it easier to catalog all the catches in one regex.

`Panic.catch (Any|JException)`

**Review by GregoryTravis — APPROVED** [review]

File: distribution/lib/Standard/AWS/0.0.0-dev/src/Database/Redshift/Redshift_Details.enso (hunk 1 above)
Comment (jdunkerley) [reply distribution/lib/Standard/AWS/0.0.0-dev/src/Database/Redshift/Redshift_Details.enso:40]: its not a from conversion and I don't think we can dynamically add the `:<Type>` at runtime.

```
Execution finished with an error: Could not find a conversion from `Function` to `Filter_Condition`.
```

File: distribution/lib/Standard/Table/0.0.0-dev/src/Data_Formatter.enso (hunk 2 above)
Comment (radeusgd) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Data_Formatter.enso:16]: Can't we have an analogous regex with `Java_Exception`?

`Panic.catch (Any|Java_Exception)`

Comment (jdunkerley) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Data_Formatter.enso:16]: yes just easier to have one name across the codebase - no actual big reason.

File: distribution/lib/Standard/AWS/0.0.0-dev/src/Database/Redshift/Redshift_Details.enso (hunk 1 above)
Comment (radeusgd) [reply distribution/lib/Standard/AWS/0.0.0-dev/src/Database/Redshift/Redshift_Details.enso:40]: As far as I know (@JaroslavTulach please tell me if I'm wrong), `(x:T)` is basically always equivalent to `T.from x`.

Comment (JaroslavTulach) [reply distribution/lib/Standard/AWS/0.0.0-dev/src/Database/Redshift/Redshift_Details.enso:40]: >  `(x:T)` is basically always equivalent to `T.from x`.

I don't think so. Following snippet:
```ruby
x = 1
y = x : Number
```
has nothing to do with conversions. Only when the type-check fails, the _conversions path_ activates.

**Pull request merged**